src/clincoded/static/components/hpo_term_modal.js

```diff
@@ -0,0 +1,193 @@
+import { FormMixin, Input } from '../libs/bootstrap/form';
+import ModalComponent from '../libs/bootstrap/modal';
+
+var HpoTermModal = createReactClass({
+    mixins: [RestMixin, FormMixin],
+
+    propTypes: {
+        addHpoTermButton: PropTypes.object.isRequired, // Button to trigger HPO modal
+        inElim: PropTypes.bool, // Used to differentiate between normal and "in elim" modal
+        savedHpo: PropTypes.array, // HPO saved from parent
+        savedElimHpo: PropTypes.array, // HPO in elimination saved from parent
+        passHpoToParent: PropTypes.func, // Function used to pass HPO data to respective parent
+        passElimHpoToParent: PropTypes.func // Function used to pass Elim HPO data to parent
+    },
+
+    getInitialState: function() {
+        return {
+            hpoWithTerms: [],
+            hpoElimWithTerms: []
+        };
+    },
+
+    handleModalClose() {
+        this.child.closeModal();
+    },
+
+    componentDidMount() {
+        const savedHpo = this.props.savedHpo ? this.props.savedHpo : null;
+        const savedElimHpo = this.props.savedElimHpo ? this.props.savedElimHpo : null;
+        if (savedHpo && savedHpo.length) {
+            this.setState({ hpoWithTerms: savedHpo });
+        }
+        if (savedElimHpo && savedElimHpo.length) {
+            this.setState({ hpoElimWithTerms: savedElimHpo });
+        }
+    },
+
+    saveForm(e) {
+        e.preventDefault(); e.stopPropagation();
+        let hpoWithTermsList = this.state.hpoWithTerms;
+        let hpoElimWithTermsList = this.state.hpoElimWithTerms;
+        const inElim = this.props.inElim ? this.props.inElim : false;
+        // Check if the modal has valid values to save, and if in Elim (NOT Phenotype); Modal has two use-cases
+        if (hpoWithTermsList && !inElim) {
+            hpoWithTermsList = [...new Set(hpoWithTermsList)];
+            this.props.passHpoToParent(hpoWithTermsList);
+        }
+        else if (hpoElimWithTermsList && inElim) {
+            hpoElimWithTermsList = [... new Set(hpoElimWithTermsList)];
+            this.props.passElimHpoToParent(hpoElimWithTermsList);
+        }
+        this.child.closeModal();
+    },
+
+    cancelForm(e) {
+        this.handleModalClose();
+    },
+
+    validateHpo(hpoids) {
+        const checkIds = curator.capture.hpoids(hpoids);
+        // Check HPO ID format
+        if (checkIds && checkIds.length && _(checkIds).any(function(id) { return id === null; })) {
+            // HPOID list is bad
+            this.setFormErrors('hpoid', 'Use HPO IDs (e.g. HP:0000001) separated by commas');
+        }
+        else if (checkIds && checkIds.length && !_(checkIds).any(function(id) { return id === null; })) {
+            const hpoidList = _.without(checkIds, null);
+            return hpoidList;
+        }
+    },
+
+    deleteHpo(index) {
+        const hpo = this.state.hpoWithTerms.filter((hpo, hpoIndex) => {
+            return hpoIndex !== index;
+        });
```

Comment:
Suggestion: if want to display list without duplicated hpo, need to delete all copies of the selected hpo.
        const term = this.state.hpoWithTerms[index];
        const hpo = this.state.hpoWithTerms.filter((hpo) => {
            return hpo !== term;